src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -19,94 +19,224 @@
+ * > online documentation</a>.
  */
-public class SimpleQueryStringBuilder extends QueryBuilder {
+public class SimpleQueryStringBuilder extends QueryBuilder<SimpleQueryStringBuilder> implements BoostableQueryBuilder<SimpleQueryStringBuilder> {
+    /** Name for (de-)serialization. */
     public static final String NAME = "simple_query_string";
-    private Map<String, Float> fields = new HashMap<>();
-    private String analyzer;
-    private Operator operator;
+    /** Query text to parse. */
     private final String queryText;
+    /** Boost to apply to resulting Lucene query. */
+    private float boost = 1.0f;
+    /**
+     * Fields to query against. If left empty will query default field,
+     * currently _ALL. Uses a TreeMap to hold the fields so boolean clauses are
+     * always sorted in same order for generated Lucene query for easier
+     * testing.
+     */
+    private final Map<String, Float> fieldsAndWeights = new TreeMap<>();
+    /** If specified, analyzer to use to parse the query text, using registered default if left empty. */
+    private String analyzer;
+    /** Name of the query. */
     private String queryName;
+    /** Default operator to use for linking boolean clauses. */
+    private Operator defaultOperator;
+    /** If result is a boolean query, minimumShouldMatch parameter to apply. Ignored otherwise. */
     private String minimumShouldMatch;
+    /** Any search flags to be used, ALL by default. */
     private int flags = -1;
-    private Boolean lowercaseExpandedTerms;
-    private Boolean lenient;
-    private Boolean analyzeWildcard;
-    private Locale locale;
-    static final SimpleQueryStringBuilder PROTOTYPE = new SimpleQueryStringBuilder(null);
+    /** Further search settings. */
+    private Settings settings = new Settings();
 
     /**
-     * Operators for the default_operator
+     * This instance is used to de-serialize {@link SimpleQueryStringBuilder}
+     * objects only. During de-serialization a new object will be created that
+     * holds the deserialized query object. As a result seeing the queryText the
+     * prototype is initialized with in any kind of query logs indicates an
+     * implemenation error.
      */
+    static final SimpleQueryStringBuilder PROTOTYPE = new SimpleQueryStringBuilder(
+            "You should never read this query in your logs unless there is an implementation error in ES.");
+
+    /** Operators available for linking boolean clauses. */
     public static enum Operator {
         AND,
         OR
     }
 
+    /** Construct a new simple query with this query string. */
+    public SimpleQueryStringBuilder(String queryText) {
+        Preconditions.checkNotNull(queryText);
+        this.queryText = queryText;
+    }
+
     /**
-     * Construct a new simple query with the given text
+     * For serialization only: Construct a new simple query string query.
```

Comment:
When serialising the flags are stored as one int instead of trying to figure out the actual fields set and storing the enum contents.

When de-serialising we need some way of setting flags based on an int. In a previous PR IIRC I had a setter method flags(int flags) and (alternatively) the constructor above. I don't remember why but I ended up removing the flags(int flags) setter and leaving the long constructor in. Should I switch that the other way around (remove the constructor and get the setter for flags provided as one int back in)?